### ISSUE TITLE
Clicking stepper arrows on number input outside Slate crashes in Firefox

### ISSUE THREAD

**Issue by bjoerge** [ianstormtaylor/slate#1819]
This seems to be related to #1744, but doesn't look like it's exactly the same issue.

![slate-bug](https://user-images.githubusercontent.com/876086/39630142-bebe2bc4-4fae-11e8-91db-c8aa746d1d2c.gif)

Here's a jsfiddle to reproduce: https://jsfiddle.net/cmu57eym/5

Happens in Firefox 59.0.2 on macOS 10.13.4 and possible in other firefoxes too.

**Comment by bjoerge** [comment]
So did some more digging and boiled it down to the following repro case (which is, from what I understand from reading the source code, is equivalent with what happens in Slate):

```html
<input type="number" id="input">
<script>
document.getElementById('input').oninput = () => {
  window.getSelection().anchorNode.nodeType
}
</script>
```

This fails with the same error as shown above.

This looks very much like a bug with Firefox. What happens is that when the `input`-event is firing, the `window.getSelection().anchorNode` is pointing to the actual stepper button, which is restricted, causing attempts to access *any* of its properties to throw.

**Comment by bjoerge** [comment]
Fixed in #2178
